cli/client_test.go

```diff
@@ -43,33 +43,46 @@ func (tw *testWriter) Write(b []byte) (int, error) {
 	return len(b), nil
 }
 
+// populateFlags populates a FlagSet from a map.
+func populateFlags(m map[string]any) *flag.FlagSet {
+	flags := &flag.FlagSet{}
+	// init all the default flags from the input
+	for name, val := range m {
+		switch v := val.(type) {
+		case int:
+			flags.Int(name, v, "")
+		case string:
+			flags.String(name, v, "")
+		case bool:
+			flags.Bool(name, v, "")
+		default:
+			// non-int and non-string flags not yet supported
+			continue
+		}
+	}
+	return flags
+}
+
+func newTestContext(t *testing.T, flags map[string]any) *cli.Context {
+	t.Helper()
+	out := &testWriter{}
+	errOut := &testWriter{}
+	return cli.NewContext(NewApp(out, errOut), populateFlags(flags), nil)
+}
+
 // setup creates a new cli.Context and viamClient with fake auth and the passed
 // in AppServiceClient and DataServiceClient. It also returns testWriters that capture Stdout and
 // Stdin.
 func setup(
 	asc apppb.AppServiceClient,
 	dataClient datapb.DataServiceClient,
 	buildClient buildpb.BuildServiceClient,
-	defaultFlags *map[string]any,
+	defaultFlags map[string]any,
```

Comment:
Did you have to switch to a regular map for some reason I'm not seeing, or just thought passing by value was better than pointer-passing here?

Reply:
I thought maps were [already pointers](https://go.dev/blog/maps) -- extra indirection seemed extra (I could be wrong about this)

Reply:
runtime/map.go:
```
// makemap implements Go map creation for make(map[k]v, hint).
// If the compiler has determined that the map or the first bucket
// can be created on the stack, h and/or bucket may be non-nil.
// If h != nil, the map can be created directly in h.
// If h.buckets != nil, bucket pointed to can be used as the first bucket.
func makemap(t *maptype, hint int, h *hmap) *hmap {

```